packages/block-library/src/navigation/style.scss

```diff
@@ -19,25 +24,192 @@
 	margin: 0;
 	padding-left: 0;
 
-	// Horizontal layout
-	display: flex;
-	flex-wrap: wrap;
+	// Only hide the menu by default if responsiveness is active.
+	.is-responsive {
+		display: none;
+	}
 
-	// Vertical layout
-	.is-vertical & {
-		display: block;
+	@include break-small() {
+		// Horizontal layout
+		display: flex;
+		flex-wrap: wrap;
+
+		// Vertical layout
+		.is-vertical & {
+			display: block;
+		}
 	}
 }
 
 // Justification.
-.items-justified-center > ul {
+// These target the named container class to work even with the additional mobile menu containers.
+.items-justified-center .wp-block-navigation__container {
 	justify-content: center;
 }
 
-.items-justified-right > ul {
+.items-justified-right .wp-block-navigation__container {
 	justify-content: flex-end;
 }
 
-.items-justified-space-between > ul {
+.items-justified-space-between .wp-block-navigation__container {
 	justify-content: space-between;
 }
+
+
+/**
+ * Mobile menu.
+ */
+
+.wp-block-navigation__responsive-container {
+	display: none;
+	position: fixed;
+	top: 0;
+	left: 0;
+	right: 0;
+	bottom: 0;
+	background-color: #fff;
+	/*
+		arbitrary, high z-index
+		@todo: find correct value in order to make sure
+		that this is always on top.
+	*/
+	z-index: 1;
+	align-items: flex-start;
+	justify-content: flex-start;
+
+	// Provide an opinionated default style for menu items inside.
+	// Inherit as much as we can regarding colors, fonts, sizes,
+	// but otherwise provide a baseline.
+	// In a future version, we can explore more customizability.
+	&.is-menu-open {
+		display: flex;
+		flex-direction: column;
+
+		// Give it a z-index just higher than the adminbar.
+		z-index: 100000;
+
+		padding: 24px;
+		background-color: inherit;
+
+		.wp-block-navigation__container {
+			display: flex;
+			flex-direction: column;
+			max-width: 400px;
+			margin-left: auto;
+			margin-right: auto;
+			align-items: flex-start;
+			line-height: 48px;
+			padding: 48px 0;
+
+			.wp-block-page-list {
+				flex-direction: column;
+			}
+		}
+	}
+
+	@include break-small() {
+		display: flex;
+		flex-direction: row;
+		position: relative;
+		background-color: inherit;
+
+		&.is-menu-open {
+			flex-direction: row;
+
+			.wp-block-navigation__container {
+				flex-direction: row;
+			}
+		}
+	}
+}
+
+// Default menu background and font color.
+.wp-block-navigation:not(.has-background) .wp-block-navigation__responsive-container {
+	background-color: #fff;
+	color: #000;
+}
+
+// Hide empty paragraph tags caused by wpautop.
+// @todo, need to not output them in the first place.
```

Comment:
Any more context on this issue? I thought block content doesn't go through the autop transformation.

https://github.com/WordPress/wordpress-develop/blob/7ef4f7c19746d2589f8875b26d5b64bac5b57025/src/wp-includes/blocks.php#L842

We do have `import { removep, autop } from '@wordpress/autop';` but typically this is used to help transform classic content.

Reply:
This is from me, and I'm not aware of any autop going on. I just see, now and then so far with no clear pattern, empty paragraphs all over the rendered page (`<p></p>`).

Reply:
@jasmussen Do any of the paragraphs show up in editor code view?

Reply:
I looked at it a few days ago and I'm pretty sure that no, everything is fine in the editor view, the extra paragraphs appear _only_ on the frontend. I did try to muck around with the whitespace in the PHP file to see if that changed anything, it did not appear to. Then again I've had npm run dev issues lately, so it's been a little harder to see what sticks and what doesn't.

Reply:
And by "all over the rendered page", I recall 3 or 4 such paragraphs.

Reply:
Maybe add an example expected/actual, html snippet? I can help step-though if this is hard to spot.

Reply:
Or course I can't reproduce them right now. I'll try and look next week to see if there's a pattern to when vs. when they don't show up.